supervisor/api/deprecation.py

```diff
@@ -0,0 +1,83 @@
+"""Handle deprecation of this API."""
+import datetime
+import logging
+
+from aiohttp.web import Request, RequestHandler, Response, middleware
+
+from ..coresys import CoreSys, CoreSysAttributes
+from .utils import excract_supervisor_token
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+DEFAULT_GRACE_PERIOD = datetime.timedelta(seconds=10)
+
+DEPRECATED_PATHS = {
+    "/hardware/trigger": {
+        "grace_period": datetime.timedelta(hours=1),
+    }
+}
+
+
+class DeprecationMiddleware(CoreSysAttributes):
+    """Deprecation middleware functions."""
+
+    def __init__(self, coresys: CoreSys):
+        """Initialize security middleware."""
+        self.coresys: CoreSys = coresys
+        self._trottle = {}
+
+    @middleware
+    async def check_deprecation(
+        self, request: Request, handler: RequestHandler
+    ) -> Response:
+        """Check if the requested path is deprecated."""
+        if request.path in DEPRECATED_PATHS:
+            now = datetime.datetime.now()
+            if request.path not in self._trottle or (
+                now - self._trottle[request.path]
+            ) > DEPRECATED_PATHS[request.path].get(
+                "grace_period", DEFAULT_GRACE_PERIOD
+            ):
+                self._trottle[request.path] = now
+                self.write_log(request)
+
+        return await handler(request)
+
+    def write_log(self, request: Request):
+        """Write the log entry."""
+        request_from = "unknown"
+        more_info = ""
+
+        supervisor_token = excract_supervisor_token(request)
```

Comment:
You don't need do that, you can just access to the data before which other middleware have generated

Reply:
Maybe just a simple function decorator would also do it instead to have it as middleware on each call?

Reply:
Sure